lib/rust/web/src/lib.rs

```diff
@@ -543,153 +628,168 @@ pub fn forward_panic_hook_to_console() {
-    target
-        .clone()
-        .dyn_into::<js_sys::Object>()
-        .ok()
-        .map(|obj| {
-            js_sys::Object::keys(&obj)
-                .iter()
-                .map(|key| {
-                    // The unwrap is safe, the `Object::keys` API guarantees it.
-                    let js_str = key.dyn_into::<js_sys::JsString>().unwrap();
-                    js_str.into()
-                })
-                .collect()
-        })
-        .unwrap_or_default()
-}
 
 
+// ====================
+// === TimeProvider ===
+// ====================
 
-// ============
-// === Test ===
-// ============
+/// Trait for an entity that can retrieve current time.
+pub trait TimeProvider {
+    /// Returns current time, measured in milliseconds.
+    fn now(&self) -> f64;
+}
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+impl TimeProvider for Performance {
+    fn now(&self) -> f64 {
+        self.now()
+    }
+}
 
-    use wasm_bindgen_test::wasm_bindgen_test;
-    use wasm_bindgen_test::wasm_bindgen_test_configure;
 
-    wasm_bindgen_test_configure!(run_in_browser);
 
-    #[cfg(target_arch = "wasm32")]
-    mod helpers {
-        type Instant = f64;
+// =================================
+// === Generic String Conversion ===
+// =================================
 
-        pub fn now() -> Instant {
-            super::performance().now()
-        }
+// FIXME: is this not the same as String::from?
 
-        pub fn elapsed(instant: Instant) -> f64 {
-            super::performance().now() - instant
-        }
-    }
+#[cfg(target_arch = "wasm32")]
+#[wasm_bindgen]
+extern "C" {
+    #[allow(unsafe_code)]
+    #[wasm_bindgen(js_name = "String")]
+    fn js_to_string_inner(s: &JsValue) -> String;
+}
 
-    #[cfg(not(target_arch = "wasm32"))]
-    mod helpers {
-        use std::time::Instant;
+#[cfg(target_arch = "wasm32")]
+/// Converts given `JsValue` into a `String`. Uses JS's `String` function,
+/// see: https://www.w3schools.com/jsref/jsref_string.asp
+pub fn js_to_string(s: impl AsRef<JsValue>) -> String {
+    js_to_string_inner(s.as_ref())
+}
 
-        pub fn now() -> Instant {
-            Instant::now()
-        }
+#[cfg(not(target_arch = "wasm32"))]
+pub fn js_to_string(_: impl AsRef<JsValue>) -> String {
+    "JsValue".into()
+}
 
-        pub fn elapsed(instant: Instant) -> f64 {
-            instant.elapsed().as_secs_f64()
-        }
-    }
 
-    #[wasm_bindgen_test(async)]
-    async fn async_sleep() {
-        let instant = helpers::now();
-        sleep(Duration::new(1, 0)).await;
-        assert!(helpers::elapsed(instant) >= 1.0);
-        sleep(Duration::new(2, 0)).await;
-        assert!(helpers::elapsed(instant) >= 3.0);
-    }
 
-    #[test]
-    #[cfg(not(target_arch = "wasm32"))]
-    fn async_sleep_native() {
-        async_std::task::block_on(async_sleep())
-    }
-}
+// ============
+// === Test ===
+// ============
+
+// #[cfg(test)]
+// mod tests {
+//     use super::*;
+//
```

Comment:
Commented block of code

Reply:
These tests were just testing std lib function for sleeping. Do not make sense. removed.